### PR TITLE
Ltjn

### PR DESCRIPTION
**Pull request by LeonardLT**
login /register/
index
main
-----first commit-----

### CODE REVIEW

**Comment by xiayu0321** [comment]
:+1:

File: app/server.js [hunk 1]
```diff
@@ -3,10 +3,16 @@ import webpackConfig from '../webpack.config';
...
 import express from 'express';
+import bodyParser from "body-parser";
+
+var routes = require('../db/loginAndRegister.js');
 
 const app = express();
 const compiler = webpack(webpackConfig);
 
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
```
Comment (LeonardLT) [inline app/server.js:14]: ？？？？？？为什么要这一行

File: app/server.js [hunk 2]
```diff
@@ -3,10 +3,16 @@ import webpackConfig from '../webpack.config';
 import webpackDevMiddleware from 'webpack-dev-middleware';
 import webpackHotMiddleware from 'webpack-hot-middleware';
 import express from 'express';
+import bodyParser from "body-parser";
+
+var routes = require('../db/loginAndRegister.js');
```
Comment (LeonardLT) [inline app/server.js:8]: login-register

Comment (LeonardLT) [inline app/server.js:8]: 把不同层的都行放在一个文件里

File: db/loginAndRegister.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+/**
```
Comment (LeonardLT) [inline db/loginAndRegister.js:1]: delete

File: db/loginAndRegister.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ * Created by Leonard on 16/8/13.
+ */
+var mongoose = require('mongoose');
```
Comment (LeonardLT) [inline db/loginAndRegister.js:4]: const

File: db/loginAndRegister.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+
+var User = mongoose.model('User', {
```
Comment (LeonardLT) [inline db/loginAndRegister.js:7]: const

File: db/loginAndRegister.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    name: String,
+    password: String,
+    email:String,
```
Comment (LeonardLT) [inline db/loginAndRegister.js:10]: 格式化代码

File: db/loginAndRegister.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    phone:Number,
```
Comment (LeonardLT) [inline db/loginAndRegister.js:11]: Number ==》String

File: db/loginAndRegister.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+});
+
+mongoose.connect("mongodb://localhost/freetravel-db");
```
Comment (LeonardLT) [inline db/loginAndRegister.js:14]: 放到server.js里，：3000之后

File: db/loginAndRegister.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+exports.insert = function (request, response) {
```
Comment (LeonardLT) [inline db/loginAndRegister.js:16]: 换成export关键字

req,res

File: db/loginAndRegister.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+    let name = request.body.name;
```
Comment (LeonardLT) [inline db/loginAndRegister.js:17]: const

File: db/loginAndRegister.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    let password = request.body.password;
+    let email = request.body.email;
+    let phone = request.body.phone;
+    User.findOne({name: name}, function (e, docs) {
```
Comment (LeonardLT) [inline db/loginAndRegister.js:21]: e =>err

File: db/loginAndRegister.js (hunk 9 above)
Comment (LeonardLT) [inline db/loginAndRegister.js:16]: function(req, res, next)

File: db/loginAndRegister.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+        if (e) response.send(e.message);
```
Comment (LeonardLT) [inline db/loginAndRegister.js:22]: if(err) return next(err)

File: db/loginAndRegister.js [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+
+        if (docs == null) {
+            console.log(name, password,email,phone);
+            var user = new User({
```
Comment (LeonardLT) [inline db/loginAndRegister.js:26]: const

File: db/loginAndRegister.js [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+    phone:Number,
+});
+
+mongoose.connect("mongodb://localhost/freetravel-db");
+
+exports.insert = function (request, response) {
+    let name = request.body.name;
+    let password = request.body.password;
+    let email = request.body.email;
+    let phone = request.body.phone;
```
Comment (LeonardLT) [inline db/loginAndRegister.js:20]: const userData = request.body

File: db/loginAndRegister.js [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+        if (e) response.send(e.message);
+
+        if (docs == null) {
+            console.log(name, password,email,phone);
+            var user = new User({
+                name: name,
+                password: password,
+                email:email,
+                phone:phone
+            });
```
Comment (LeonardLT) [inline db/loginAndRegister.js:31]: new User(userData)

File: db/loginAndRegister.js [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+            user.save(function (err) {
+                console.log('save status:', err ? 'failed' : 'success');
```
Comment (LeonardLT) [inline db/loginAndRegister.js:33]: if(err) return next(err)

File: db/loginAndRegister.js [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+                response.send('success');
+
+            });
+        }
+        else {
+        //
+        response.send('error');
+        // var tpl = '<script>alert("用户已存在")</script>';
+        // response.send(tpl); //你传回去的HTML码流将在客户的浏览器中执行
+        // response.redirect('./signin');
```
Comment (LeonardLT) [inline db/loginAndRegister.js:43]: remove commented code

File: db/loginAndRegister.js [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+            });
+            user.save(function (err) {
+                console.log('save status:', err ? 'failed' : 'success');
+                response.send('success');
+
+            });
+        }
+        else {
+        //
+        response.send('error');
```
Comment (LeonardLT) [inline db/loginAndRegister.js:40]: response.status(500).send('error')

File: db/loginAndRegister.js [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+        // var tpl = '<script>alert("用户已存在")</script>';
+        // response.send(tpl); //你传回去的HTML码流将在客户的浏览器中执行
+        // response.redirect('./signin');
+        }
+    });
+
+};
+
+exports.login = function (request, response) {
```
Comment (LeonardLT) [inline db/loginAndRegister.js:49]: export

Comment (LeonardLT) [inline db/loginAndRegister.js:49]: next

File: db/loginAndRegister.js [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+    let name = request.body.name;
+    let password = request.body.password;
+    console.log('---finding---');
+    User.findOne({name: name, password: password}, function (e, docs) {
+        if (e) response.send(e.message);
```
Comment (LeonardLT) [inline db/loginAndRegister.js:54]: if(err) return next(err)

File: db/loginAndRegister.js [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+
+        if (docs != null) {
+            // response.location('https://www.baidu.com');
+            // response.redirect('./');
+            response.send('success');
+            // response.send(docs+'<br><br>login success');
+        } else {
+            // response.redirect('./login');
+            response.send('error');
```
Comment (LeonardLT) [inline db/loginAndRegister.js:63]: 500

File: db/loginAndRegister.js [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+exports.login = function (request, response) {
+    let name = request.body.name;
+    let password = request.body.password;
+    console.log('---finding---');
+    User.findOne({name: name, password: password}, function (e, docs) {
+        if (e) response.send(e.message);
+
+        if (docs != null) {
```
Comment (LeonardLT) [inline db/loginAndRegister.js:56]: note

File: db/loginAndRegister.js [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+            // response.redirect('./login');
+            response.send('error');
+
+        }
+    });
+
+};
+
+exports.remove = function(request, response) {
+    // mongoose.connect("mongodb://localhost/mongodb-find-demo-db");
```
Comment (LeonardLT) [inline db/loginAndRegister.js:71]: delete

File: db/loginAndRegister.js [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+    let name = request.body.name;
```
Comment (LeonardLT) [inline db/loginAndRegister.js:72]: const

File: db/loginAndRegister.js [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+    let password = request.body.password;
+    User.remove({
+        name: name,
+        password:password
+    }, function(e) {
+        if (e) response.send(e.message);
```
Comment (LeonardLT) [inline db/loginAndRegister.js:78]: if(err) return next(err)

File: db/loginAndRegister.js [hunk 26]
```diff
@@ -0,0 +1,81 @@
...
+            console.log(name, password,email,phone);
+            var user = new User({
+                name: name,
+                password: password,
+                email:email,
+                phone:phone
+            });
+            user.save(function (err) {
+                console.log('save status:', err ? 'failed' : 'success');
+                response.send('success');
```
Comment (LeonardLT) [inline db/loginAndRegister.js:34]: response.sendStatus(201)

File: public/brief/brizyn.jsx [hunk 27]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import rent from "./icon/rent.png";
+import free from "./icon/free.png";
+import friend from './icon/friend.png';
+// require('jquery');
+// require("bootstrap-webpack");
```
Comment (LeonardLT) [inline public/brief/brizyn.jsx:6]: delete

Comment (LeonardLT) [inline public/brief/brizyn.jsx:6]: delete

File: public/brief/brizyn.jsx [hunk 28]
```diff
@@ -0,0 +1,38 @@
...
+require("./csszyn.css");
```
Comment (LeonardLT) [inline public/brief/brizyn.jsx:7]: zyn

File: public/brief/brizyn.jsx [hunk 29]
```diff
@@ -0,0 +1,38 @@
...
+
+class Brife extends React.Component {
```
Comment (LeonardLT) [inline public/brief/brizyn.jsx:9]: a better name?

File: public/brief/csszyn.css [hunk 30]
```diff
@@ -0,0 +1,9 @@
+.summary {
+    padding-right: 15px;
+    padding-left: 15px;
+}
+.summary .col-md-4 {
+    margin-bottom: 20px;
+    text-align: center;
+}
+
```
Comment (LeonardLT) [inline public/brief/csszyn.css:9]: filename

File: public/brief/brizyn.jsx [hunk 31]
```diff
@@ -0,0 +1,38 @@
...
+    render() {
+        return (
+            <div className="container summary">
+                <div className="row" id="summary-container">
+                    <div className="col-md-4">
```
Comment (LeonardLT) [inline public/brief/brizyn.jsx:14]: channel

File: public/brief/csszyn.css (hunk 30 above)
Comment (LeonardLT) [inline public/brief/csszyn.css:9]: .summary .channel

File: public/entry.jsx [hunk 32]
```diff
@@ -12,10 +13,12 @@ require("bootstrap-webpack");
 
 
 const router = <Router history={hashHistory}>
-    <Route path='/' component={App}>
+    <Route path="/" component={Index}>
+
+    </Route>
+    <Route path='/main' component={App}>
         <IndexRoute component={Hello}/>
```
Comment (LeonardLT) [inline public/entry.jsx:20]: <IndexRediect to='/hello' />

File: public/index.jsx [hunk 33]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Created by Leonard on 16/8/13.
+ */
```
Comment (LeonardLT) [inline public/index.jsx:3]: rm

File: public/index.jsx [hunk 34]
```diff
@@ -0,0 +1,41 @@
...
+import React from 'react';
+import Brife from './brief/brizyn.jsx';
+import Pic from './pictureswall/piczyn.jsx';
```
Comment (LeonardLT) [inline public/index.jsx:6]: name

File: public/index.jsx [hunk 35]
```diff
@@ -0,0 +1,41 @@
...
+                            <TabsNavs/>
+                        </div>
+                        <div className="pull-right loginNavs">
+                            <Navs/>
+                        </div>
+                    </div>
+                    {/*/!*body*!/*/}
+                    {/*<div>*/}
+                    {/*{this.props.children}*/}
+                    {/*</div>*/}
```
Comment (LeonardLT) [inline public/index.jsx:28]: remove comments

File: public/pictureswall/piczyn.jsx [hunk 36]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
```
Comment (LeonardLT) [inline public/pictureswall/piczyn.jsx:1]: pictures-wall

File: public/pictureswall/piczyn.jsx [hunk 37]
```diff
@@ -0,0 +1,74 @@
...
+import beauty from './pictrue/beauty.jpeg';
```
Comment (LeonardLT) [inline public/pictureswall/piczyn.jsx:2]: beautyImage

File: public/register/register-qufan.jsx [hunk 38]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
 import './register-qufan.css'
 
 class Register extends React.Component{
+
+    commit() {
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:6]: submit

File: public/register/register-qufan.jsx [hunk 39]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+        let name = $('#name').val();
+        let password = $('#password').val();
+        let email = $('#email').val();
+        let phone = $('#phone').val();
+        $.ajax({
+            url: './register',
+            type: 'POST',
+            async: true,
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:14]: unnecessary

File: public/register/register-qufan.jsx [hunk 40]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
 import './register-qufan.css'
 
 class Register extends React.Component{
+
+    commit() {
+        let name = $('#name').val();
+        let password = $('#password').val();
+        let email = $('#email').val();
+        let phone = $('#phone').val();
+        $.ajax({
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:11]: $.post

File: public/register/register-qufan.jsx [hunk 41]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+            url: './register',
+            type: 'POST',
+            async: true,
+            contentType: 'application/json',
+            data: JSON.stringify({name: name, password: password,email:email,phone:phone}),
+            // data: {name :name,password:password},
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:17]:
```
type: 'json',
data: {name: name, ..}
```

File: public/register/register-qufan.jsx [hunk 42]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+    commit() {
+        let name = $('#name').val();
+        let password = $('#password').val();
+        let email = $('#email').val();
+        let phone = $('#phone').val();
+        $.ajax({
+            url: './register',
+            type: 'POST',
+            async: true,
+            contentType: 'application/json',
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:15]: remove

File: public/register/register-qufan.jsx [hunk 43]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+            data: JSON.stringify({name: name, password: password,email:email,phone:phone}),
+            // data: {name :name,password:password},
+
+            success(result) {
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:19]: success: function(result) {}

File: public/register/register-qufan.jsx [hunk 44]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+//                    $("#div1").html(result);
+                if(result=='success'){
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:21]: don't need to check 'success'

File: public/register/register-qufan.jsx [hunk 45]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+                    alert('success');
+                    // location.href='./rent'
+                    self.location='/#/rent'
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:24]: ???

File: public/register/register-qufan.jsx [hunk 46]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+                }
+                else if(result == "error") {
+                    // location.href='#/'
+                    alert('用户已存在');
+                }
+            },
+            error(result){
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:31]: error: function() {}

File: public/register/register-qufan.jsx [hunk 47]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+            // data: {name :name,password:password},
+
+            success(result) {
+//                    $("#div1").html(result);
+                if(result=='success'){
+                    alert('success');
+                    // location.href='./rent'
+                    self.location='/#/rent'
+                }
+                else if(result == "error") {
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:26]: should not be here

File: public/register/register-qufan.jsx [hunk 48]
```diff
@@ -2,10 +2,46 @@ import React from 'react';
...
+                    // location.href='#/'
+                    alert('用户已存在');
+                }
+            },
+            error(result){
+                if (result == "error") {
+                    // location.href='#/'
+                    alert('用户已存在');
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:34]: ???

File: public/register/register-qufan.jsx [hunk 49]
```diff
@@ -32,9 +68,9 @@ class Register extends React.Component{
                         <input type="password" className="form-control" id="confirm-password"
                                placeholder="请确认密码" required pattern="^.{6,18}$"/>
                     </div>
-                    <input type="submit" value="注册" className="btn btn-primary"/>
+                    <input type="submit" value="注册" className="btn btn-primary"  onClick={this.commit}/>
```
Comment (LeonardLT) [inline public/register/register-qufan.jsx:71]: this.commit.bind(this)

File: public/signIn/css/signIn.css [hunk 50]
```diff
@@ -0,0 +1,138 @@
+body{
```
Comment (LeonardLT) [inline public/signIn/css/signIn.css:1]: filename: sign-in

File: public/signIn/jsx/signIn.jsx [hunk 51]
```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import '../css/signIn.css';
```
Comment (LeonardLT) [inline public/signIn/jsx/signIn.jsx:2]: sign-in

File: public/signIn/jsx/signIn.jsx [hunk 52]
```diff
@@ -0,0 +1,71 @@
...
+
+class SignIn extends React.Component {
+    login() {
+        let name = $('#name').val();
+        let password = $('#password').val();
```
Comment (LeonardLT) [inline public/signIn/jsx/signIn.jsx:7]: *

File: public/signIn/jsx/signIn.jsx [hunk 53]
```diff
@@ -0,0 +1,71 @@
...
+                    alert('用户-存在');
+                }
+            }
+        });
+
+    }
+
+    render() {
+        return (<div>
+            <div className="top_div"></div>
```
Comment (LeonardLT) [inline public/signIn/jsx/signIn.jsx:40]: top?

File: public/signIn/jsx/signIn.jsx [hunk 54]
```diff
@@ -0,0 +1,71 @@
...
+            <div className="cont">
+                <div className="onePart">
+                    <div className="tou"></div>
+                    <div className="initial_left_hand" id="left_hand"></div>
```
Comment (LeonardLT) [inline public/signIn/jsx/signIn.jsx:44]: ##

File: public/signIn/jsx/signIn.jsx [hunk 55]
```diff
@@ -0,0 +1,71 @@
...
+            }
+        });
+
+    }
+
+    render() {
+        return (<div>
+            <div className="top_div"></div>
+            <div className="cont">
+                <div className="onePart">
```
Comment (LeonardLT) [inline public/signIn/jsx/signIn.jsx:42]: one-part

File: public/signIn/jsx/signIn.jsx [hunk 56]
```diff
@@ -0,0 +1,71 @@
...
+                    <div className="tou"></div>
```
Comment (LeonardLT) [inline public/signIn/jsx/signIn.jsx:43]: tou?

File: public/signIn/jsx/signIn.jsx [hunk 57]
```diff
@@ -0,0 +1,71 @@
...
+                    <div className="initial_left_hand" id="left_hand"></div>
+                    <div className="initial_right_hand" id="right_hand"></div>
+                </div>
+                <p className="pipe">
+                    <span className="u_logo"></span>
+                    <input className="ipt" type="text" placeholder="请输入用户名或邮箱" id="name" />
```
Comment (LeonardLT) [inline public/signIn/jsx/signIn.jsx:49]: ipt?

File: public/signIn/jsx/signIn.jsx [hunk 58]
```diff
@@ -0,0 +1,71 @@
...
+                    <input className="ipt" id="password" type="password" placeholder="请输入密码" />
+                </p>
+                <div className="forget">
+                    <p className="setMargin">
+                        <span className="setPos" >
+                            <a className="setColor">忘记密码?</a>
+                        </span>
+                        <span className="setPos">
+                            <a className="setColor setM" >注册</a>
+                            <a className="signIn" onClick={this.login}>登录</a>
```
Comment (LeonardLT) [inline public/signIn/jsx/signIn.jsx:62]: this.login.bind(this)

**Pull request merged**